### PR TITLE
feat(server): fix the metrics middleware

### PR DESCRIPTION
**Pull request by bojiang**
## Description

The solution of the OpenTelemetry Prometheus exporter is still not ready for production (`alpha`, otel said).
Let's focus on `promethrus_client`.

TODO:
- [x] maintain multiproc dir
- [ ] ci checking of metrics content

## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/bento/bento.py [hunk 1]
```diff
@@ -168,7 +168,8 @@ def create(
         relpaths = [s for s in build_config.include if s.startswith("../")]
         if len(relpaths) != 0:
             raise InvalidArgument(
-                "Paths outside of the build context directory cannot be included; use a symlink or copy those files into the working directory manually."
```
Comment (parano) [inline bentoml/_internal/bento/bento.py:171]: use one line error message to make it easy to search

File: bentoml/_internal/bento/bento.py [hunk 2]
```diff
@@ -168,7 +168,8 @@ def create(
...
+                "Paths outside of the build context directory cannot be included; use "
+                "a symlink or copy those files into the working directory manually."
```
Comment (parano) [inline bentoml/_internal/bento/bento.py:172]:
```suggestion
                "Paths outside of the build context directory cannot be included; use a symlink or copy those files into the working directory manually."
```

File: bentoml/_internal/server/metrics/prometheus.py [hunk 3]
```diff
@@ -20,87 +21,111 @@ def __init__(
...
-        os.makedirs(multiproc_dir, exist_ok=True)
+    @property
+    def prometheus_client(self):
+        if self.multiproc and not self._imported:
+            # step 1: check environment
+            assert (
+                "prometheus_client" not in sys.modules
+            ), "prometheus_client is already imported, multiprocessing will not work properly"
+
+            assert self.multiproc_dir, repr(self.multiproc_dir)
```
Comment (parano) [inline bentoml/_internal/server/metrics/prometheus.py:45]:
```suggestion
            assert self.multiproc_dir, f"Invalid prometheus multiproc directory: {self.multiproc_dir}"
```

File: bentoml/_internal/server/__init__.py [hunk 4]
```diff
@@ -109,6 +124,7 @@ def serve_development(
     )
 
     arbiter = create_standalone_arbiter(watchers)
+    _ensure_prometheus_dir()
```
Comment (parano) [inline bentoml/_internal/server/__init__.py:127]: 👍 great job refactoring this, it's clear this is an one time setup and not invoked in child process

**Comment by aarnphm** [comment]
i failed because it actually doesn't reach the code coverage :)

**Pull request merged**